lms/djangoapps/course_api/blocks/permissions.py

```diff
@@ -30,5 +30,6 @@ def can_access_self_blocks(requesting_user, course_key):
     """
     return (
         (requesting_user.id and CourseEnrollment.is_enrolled(requesting_user, course_key)) or
-        has_access(requesting_user, CourseStaffRole.ROLE, course_key)
+        has_access(requesting_user, CourseStaffRole.ROLE, course_key) or
+        is_course_public(get_course_by_id(course_key, depth=0))
```

Comment:
@xitij2000, `get_course_by_id()` will raise a `Http404` exception and that is causing [this unit test failure](https://build.testeng.edx.org/job/edx-platform-python-pipeline-pr/18734/testReport/junit/lms.djangoapps.course_api.blocks.tests.test_forms/TestBlockListGetForm/Run_Tests___lms_unit___test_non_existent_usage_key/). Should we wrap that call in a `try-except` call and return a boolean based on what happens?